relay-server/src/actors/envelopes.rs

```diff
@@ -361,9 +364,15 @@ impl EnvelopeManagerService {
         envelope.add_item(item);
 
         let partition_key = partition_key.map(|x| x.to_string());
-        self.submit_envelope(envelope, scoping, partition_key)
+        match self
+            .submit_envelope(envelope, scoping, partition_key)
             .await
             .map_err(|_| buckets)
+        {
+            Ok(()) => {}
+            Err(buckets) => Aggregator::from_registry()
+                .do_send(relay_metrics::MergeBuckets::new(project_key, buckets)),
+        }
```

Comment:
Same as above. Additionally, the `map_err` can be omitted and I would propose to import `MergeBuckets`:

```suggestion
        let result = self
            .submit_envelope(envelope, scoping, partition_key)
            .await;

        match result {
            Ok(()) => (),
            Err(_) => Aggregator::from_registry()
                .do_send(MergeBuckets::new(project_key, buckets)),
        }
```

Reply:
addressed in [498277e](https://github.com/getsentry/relay/pull/1492/commits/498277ea80d283500dd24e56d90a3b2fd2c6b3ed)